src/tests/tests/background.js

```diff
@@ -139,4 +178,93 @@
 
     badger.isCheckingDNTPolicyEnabled = old_dnt_check_func;
   });
+
+  QUnit.module("tabData", {
+    beforeEach: function () {
+      this.tabId = -1;
+      badger.tabData[this.tabId] = {
+        frames: {},
+        origins: {},
+        blockedCount: 0
+      };
+    },
+  },
+  function() {
+    QUnit.module("logThirdPartyOriginOnTab", {
+      beforeEach: function () {
+        this.before = beforeMock(['badger.updateBadge']);
+        badger.updateBadge = noop;
+      },
+      afterEach: function () {
+        unmock(this.before);
+      },
+    });
+    QUnit.test("increment blocked count", function(assert) {
+      let tabId = this.tabId;
+      assert.equal(badger.tabData[tabId].blockedCount, 0);
+
+      badger.logThirdPartyOriginOnTab(tabId, 'stuff', constants.BLOCK);
+      assert.equal(badger.tabData[tabId].blockedCount, 1);
+
+      badger.logThirdPartyOriginOnTab(tabId, 'stuff', constants.BLOCK);
+      assert.equal(badger.tabData[tabId].blockedCount, 1);
+
+      badger.logThirdPartyOriginOnTab(tabId, 'eff.org', constants.COOKIEBLOCK);
+      assert.equal(badger.tabData[tabId].blockedCount, 2);
+    });
+    QUnit.module('updateBadge', {
+      beforeEach: function() {
+        this.before = beforeMock([
+          'chrome.tabs.get',
+          'chrome.browserAction.setBadgeText',
+          'chrome.browserAction.setBadgeBackgroundColor'
+        ]);
+      },
+      afterEach: function() {
+        unmock(this.before);
+      },
+    });
+    QUnit.test("disabled", function(assert) {
+      let done = assert.async(2),
+        called = false;
```

Comment:
We should be using [Sinon.JS](http://sinonjs.org/) for mocking/stubbing/verifying whether and how many times something got called, instead of rolling our own test helpers.